util: replace quotapool.TokenBucket with cockroachdb/quotapool

RaduBerinde opened the pull request:
I have pulled out the token bucket implementation in a separate repo (so Pebble can use it too). This commit removes it from Cockroach and refactors the code to use the library.

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/104844)
<!-- Reviewable:end -->

RaduBerinde committed:
util: replace quotapool.TokenBucket with cockroachdb/quotapool

I have pulled out the token bucket implementation in a separate repo
(so Pebble can use it too). This commit removes it from Cockroach and
refactors the code to use the library.

Epic: none
Release note: None

RaduBerinde commented:
TFTR!

bors r+
